libgalois/include/galois/ParallelSTL.h

```diff
@@ -305,6 +306,61 @@ std::enable_if_t<!std::is_scalar<internal::Val_ty<I>>::value> destroy(I first,
 template <class I>
 std::enable_if_t<std::is_scalar<internal::Val_ty<I>>::value> destroy(I, I) {}
 
+/**
+ * Does a partial sum from first -> last and writes the results to the d_first
+ * iterator.
+ */
+template <class InputIt, class OutputIt>
+OutputIt partial_sum(InputIt first, InputIt last, OutputIt d_first) {
+  using ValueType = typename std::iterator_traits<InputIt>::value_type;
+
+  size_t sizeOfVector = std::distance(first, last);
+
+  // optimization possible here: if num blocks = 1, then no need for 2 passes
```

Comment:
This is simple to enforce. Just check if sizeOfVector <= 1024 (use the same value as sort in the file), then call std::partial_sum() and return.

Reply:
ack